### ISSUE TITLE
Assess on vmap combinator can fail on Masked choicemap

### ISSUE THREAD

**Issue by georgematheos** [genjax-community/genjax#1462]
Minimal working example:
```python
gf = genjax.flip.vmap(in_axes=(0,))

@jax.jit
def get_choicemap(idx):
    return genjax.ChoiceMap.switch(
        idx=idx,
        chms=[
            C.set(jnp.array([0, 0, 1], dtype=bool)),
            C.set(jnp.array([1, 1, 1], dtype=bool)),
        ]
    )

chm = get_choicemap(1)

flipprobs = jnp.array([0.2, 0.4, 0.6])
gf.importance(PRNGKey(0), chm, (flipprobs,))
# ^ This line runs.
# However, when I try to call assess in the exact
# same configuration, I get an error.
gf.assess(chm, (flipprobs,)) # Errors.
```

The error is
```
ValueError: vmap was requested to map its argument along axis 0, which implies that its rank should be at least 1, but is only 0 (its shape is ())
```

**Comment by sritchie** [comment]
I see why you'd want to use `switch` this way… let me think about it, but for now try

```
def get_choicemap(idx, a, b):
    return genjax.ChoiceMap.switch(
        idx=idx,
        chms=[
            C.set(a, dtype=bool)),
            C.set(b, dtype=bool)),
        ]
    )
 
chm = jax.vmap(lambda a, b: get_choicemap(1, a, b))(jnp.array([0, 0, 1]), jnp.array([1,1,1]))
```

something like that. That should work for your vmap use case! I'm typing in Linear so this could be buggy…

**Comment by georgematheos** [comment]
Got it, thank you for the quick response!

The setting in my code is actually quite a bit more complicated than the MWE I shared.  I actually have a switch combinator over two static generative functions which each use vmap combinators.  As a result, choicemaps for my generative function naturally look like switch choicemaps with vector-shaped choices inside.  I then have a function which takes some other data structures as input, and constructs these "natural choicemaps" for my generative function, as a way to get choicemaps for my generative function at different points in my codebase.  As a result it won't be a super easy refactor to make a change like the one you suggested in the full codebase.  Fortunately there's still a quick workaround I've been using -- I just swapped in `importance` wherever I had been using `assess`!
